### ISSUE TITLE
Switch does not properly handle cancelling of toggle event

### ISSUE THREAD

**Issue by davepruitt** [dotnet/maui#14774]
### Description

In my app, I have a switch control. When the user attempts to toggle the switch either on or off, I want to display an alert to the user and basically ask them "Are you sure you want to do this?". If they respond "no", then I want to cancel the toggle event (return the switch to the state it was originally in).

You would think this would be fairly simple, but I have tried multiple ways of doing this, and they have all failed. Therefore, I am convinced there is a bug.

I have created a simple example and uploaded it to Github to demonstrate reproducing this bug. In this simple example, I bind the `IsToggled` value of the switch to a value on a view-model class. I use one-way binding so that toggling the switch will not automatically call the view-model's setter. When the user toggles the switch, I handle the `Toggled` event, and I display an alert to the user asking if they are sure they want to do that action. If they say "Yes", then I manually set the value on the view-model and notify the UI of the property change (using INotifyPropertyChanged).

If the user says "No", ideally the switch would just remain its original value. This could happen a couple of ways:
1. In my example code, I just notify the UI of a property change event (even though I didn't actually change the property). This causes the UI to refresh the `IsToggled` binding. By stepping through the code, I can see that indeed it _is calling the getter on the view model_, but unfortunately the UI does not reflect the actual state of the variable on the view-model.
2. Another way this could be done is by manually setting `IsToggled` on the Switch control to be the value that I want it to be (the original value before the user toggled the switch). Unfortunately, this is impossible because apparently setting `IsToggled` programmatically causes the `Toggled` event to occur, and thus forces me to end up in an endless loop.

So, neither of the above methods works properly. There is a bug in the Switch control. Please fix it. :)

### Steps to Reproduce

1. Run the app in the linked repository.
2. Try toggling the switch
3. Select "No" in the dialog to "cancel" the toggling of the switch
4. Check and see if the switch is now in the toggled or un-toggled state.

### Link to public reproduction project repository

https://github.com/davepruitt/bug_reproduction_repository_04_maui_switch

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, Windows, I was *not* able test on other platforms

### Affected platform versions

Windows 10, Android 13

### Did you find any workaround?

Nope. Any suggestions are welcome.

**Comment by davepruitt** [comment]
Adding a comment here with an _ugly_ workaround that I am currently using. I can get option (2) from my above post to work, _if and only if_ I use this ugly workaround.

In my `MainPage.xaml.cs` code, I add a private boolean variable called `infinite_loop_gate`, and then I re-define my `Toggled` event handler to do the following:
```
private bool infinite_loop_gate = false;

private async void MySwitch_Toggled(object sender, ToggledEventArgs e)
{
	if (!infinite_loop_gate)
	{
        var result = await DisplayAlert("Are you sure?", string.Empty, "Yes", "No");
        var vm = this.BindingContext as MainPageViewModel;
        if (vm != null)
        {
            if (result)
            {
                vm.IsMySwitchToggled = e.Value;
            }
            else
            {
                infinite_loop_gate = true;
                MySwitch.IsToggled = !e.Value;
            }
        }
    }
    else
    {
        infinite_loop_gate = false;
    }
}
```

This breaks the infinite loop caused by setting `IsToggled` programmatically. Despite this _ugly_ workaround, there is still a significant bug that needs to be fixed here in the .NET MAUI Switch.

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.7.0 Preview 2.0. Repro on Windows 11, Android 13.0-API33 and iOS 16.4 with below Project:
[ToggleSwitchValidationTest.zip](https://github.com/dotnet/maui/files/11945868/ToggleSwitchValidationTest.zip)
![switch](https://github.com/dotnet/maui/assets/97647558/29d58e99-4eef-4a2e-9df6-1ce5f81c82ca)

**Comment by rjspencer91** [comment]
+1 for this

Annoyingly SwitchCell in TableView seems to work fine, whereby the equivalent toggled event only fires on user interaction, and not from a data binding. Switch itself though is currently quite unusable in our project sadly, you should be able to set IsToggled value without firing the Toggled event

Is there any update on a timeline for this?

**Comment by Alex-111** [comment]
That's true - It is really bad, that we cannot distinguish if the value changed because of user interaction or because of code/bindings...

**Comment by andyzukunft** [comment]
I just discovered the same issue. I actually thought Microsoft forgot to add `e.Abort()` or something like that to the event. But your guesses might also be correct, that the event was not intented to fire if being set by code.

Anyway, currently one needs to implement a custom loop-preventer in the code behind file similar to the one proposed by davepruitt.

**Comment by WebGoose** [comment]
We are having the same problem.

Is there any update on this issue? We have also had to resort to a similar workaround.

**Comment by copycatcode** [comment]
This seems to be the same cause as #12516.

**Comment by XamMattia83** [comment]
I have the same problem (.net maui 9.0.70). As workaround, i dediced to bind a command on a "TapGestureRecognizer" insted of Toggle event.
